TT_NET_CLIENT_SIDE/TTNETAPI_Sample_Console_AlgoOrderRouting/TTNetApiFunctions.cs

```diff
@@ -98,6 +98,33 @@ public void ttNetApiInitHandler(TTAPI api,ApiCreationException ex)
             }
         }
 
+        void StartAlgo()
+        {
+            while (! m_price.IsValid || m_algo ==null)
+            {
+                Console.WriteLine("Waiting ...");
+                Thread.Sleep(1000);
```

Comment:
This will work but rather than sleeping, can you wait on an event? It is more efficient than sleeping since it happens right when you want it to happen not some time later.

Reply:
And we can leave the other sample the way it is. This one can just be better than that one :-)

Reply:
change to wait/set